Add references to the Pipeline templates onto the docs (#2741)
buildkite/docs: pages/pipelines/example_pipelines.md

# Example pipelines

> Example [pipelines](/docs/pipelines) can be found in our [pipeline template gallery](https://buildkite.com/pipelines/templates) covering a wide range of technologies (such as [javascript](https://buildkite.com/pipelines/templates?language=javascript)) and use cases (such as [infrastructure as code](https://buildkite.com/pipelines/templates?useCase=iac)).

Reviewer: ```suggestion
Example [pipelines](/docs/pipelines) can be found in our [pipeline template gallery](https://buildkite.com/pipelines/templates) covering a wide range of technologies (such as [JavaScript](https://buildkite.com/pipelines/templates?language=javascript)) and use cases (such as [infrastructure as code](https://buildkite.com/pipelines/templates?useCase=iac)).
```

Reviewer: Thanks very much for your update @clbarrell !

This minor correction should hopefully stop the linter from breaking...

> Example [pipelines](/docs/pipelines) can be found in our [pipeline template gallery](https://buildkite.com/pipelines/templates) covering a wide range of technologies (such as [JavaScript](https://buildkite.com/pipelines/templates?language=javascript)) and use cases (such as [infrastructure as code](https://buildkite.com/pipelines/templates?useCase=iac)).

Reviewer: Minor wording tweaks...

```suggestion
Example [pipelines](/docs/pipelines) can be found in Buildkite's [pipeline template gallery](https://buildkite.com/pipelines/templates), covering a wide range of technologies (for example, [JavaScript](https://buildkite.com/pipelines/templates?language=javascript)), and use cases (for example, [infrastructure as code](https://buildkite.com/pipelines/templates?useCase=iac)).
```

Replaced 'such as' with 'for example', since 'such as' implies a list of items, particularly if it's preceded by a description such as '_wide_ range of technologies'.
